### ISSUE TITLE
Feature request: unit tests with the lua_mapgen_kernel

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#6051]
AFAICS, there are no unit tests that run with the `mapgen` kernel. This is a feature request to either make that possible, or to document it if it's already possible.

Related to #5108, and trying to write a unit test to answer the "does this work?" questions in https://github.com/wesnoth/wesnoth/issues/5108#issuecomment-868894056 .

**Comment by CelticMinstrel** [comment]
Hmm. I think it might be technically _possible_ currently but not in any nice way…

Unit tests in the application kernel could be nice too. This should be relatively easy – just run them as a script on a specific unit test scenario. Then again, I'm not sure what you'd need to test there that you can't test in the game kernel, other than the stuff already covered by the MP test.

**Comment by stevecotton** [comment]
> trying to write a unit test to answer the "does this work?" questions in 5108

Progress so far is looking good. Agreed with you that it looks like an off-by-one error:
```
			local filter_hexes_to_change = wesnoth.map.filter(
				{"all",
					{"x", "8-12"},
					{"y", "2-4"}
				}
			)
```
```
Assertion failed (expected 15 == 16): castles should not have changed
Assertion failed (expected 1 == 2): keeps should not have changed
Assertion failed (expected "Xv" == "Gg^Kov"): row 5 should not have changed
FAIL TEST (UNKNOWN): mapgen_filter_range
```
![image](https://user-images.githubusercontent.com/101462/131256226-9e20dabf-340b-4f4e-97d5-bdf73e86308b.png)

**Comment by stevecotton** [comment]
Implemented a test in #6058.
